enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -122,7 +122,7 @@ export const refreshTokenAsync = createAsyncThunk(
         }
       }
       // Lastly if we don't have an error message or status, check if we actually got the session ID
-      if (!("id" in session)) {
+      if (!("id" in session) || typeof session.id !== "string") {
```

Comment:
I feel like this isn't the way we're supposed to check for the proper session.

What if we do it like

```suggestion
      if (!("id" in session && "exp" in session && "iat" in session && "status" in session)) {
```

would it be less likely to have to adjust this logic in the future?

Reply:
the reason I raise this is that our message specifies `{id:string, exp:number, iat:number, status:string}` and it'd be misleading if we only need `id` wouldn't it?

I understand your point though.

What about using some kind of pattern-matching/object validation lib for this? [I think we use `yup`](https://metaboat.slack.com/archives/C505ZNNH4/p1732017723090489?thread_ts=1732016660.975619&cid=C505ZNNH4). I'd be a lot cleaner than doing this manually even if for just 2 properties.

So sum up. What do you think about using `yup` to check for 2 properties `id` and `exp` that they're the correct types?

Reply:
Yep that's a good idea, I'll do that.

Reply:
Done in 1a143f01b433b300f93d2016b87152633b38bd70